Warzone/MapLoader.cpp

```diff
@@ -1,239 +1,30 @@
-			if (bordersIt != vec.end()) {
-				bordersStart = static_cast<int>(distance(vec.begin(), bordersIt));
-			}
-
-
-			// find end of file
-			vector<string>::iterator endOfFileIt = find(vec.begin(), vec.end(), str);
-			if (endOfFileIt != vec.end()) {
-				bordersEnd = static_cast<int>(distance(vec.begin(), endOfFileIt));
-			}
-			mapFile.close();
-			// using the starting point for countries and borders
-			// store territory objects
-
-			int j = 0;
-			for (int i = bordersStart + 1; i < bordersEnd; ++i) {
-				j++;
-				vector<string> subStr = split(vec[i], ' ');
-				for (int k = 1; k < subStr.size(); k++) {
-					territoryBordersVec.push_back(new Territory(stoi(subStr[0]), stoi(subStr[k]), territoryVec[j - 1]->getTerritoryName(), territoryVec[j - 1]->getTerritoryContinent()));
-				}
-			}
-			vector<Territory*>::iterator iter;
-			for (iter = territoryBordersVec.begin(); iter != territoryBordersVec.end(); ++iter) {
-				(*iter)->displayTerritories();
-			}
-			cout << "*** Extracting countries/territories from " << this->fileName << " map... DONE!" << endl;
-		}
-		else {
-			cout << this->fileName << " not found!" << endl;
-		}
-	}
-}
-
-vector<Continent*> MapLoader::getContinents() {
-	return continentVec;
-}
-
-vector<Territory*> MapLoader::getTerritories() {
-	return territoryVec;
-}
-
-vector<Territory*> MapLoader::getTerritoriesWithBorders() {
-	return territoryBordersVec;
-}
-
-
-// Function to return the corresponding Continent object to the continent index
-Continent MapLoader::getTerritoryContinentObject(int index) {
-	for (int i = 0; i < continentVec.size(); ++i) {
-		if (index == i + 1)
-		{
-			return *continentVec.at(i);
-		};
-	}
-}
-
-// Split function to separate a line of strings into a vector of substrings to extract specific information from a .map file line
-vector<string> split(string str, char delimiter) {
-	vector<string> internal;
-	stringstream ss(str); // Turn the string into a stream. 
-	string tok;
-
-	while (getline(ss, tok, delimiter)) {
-		internal.push_back(tok);
-	}
-
-	return internal;
+#include <string>
+#include <vector> 
+#include <iostream>
+#include "Player.h"
+#include "map.h"
+using namespace std;
+
+int main() {
+	Territory a=Territory(2, "USA", 3);
+	Territory b = Territory(3, "Canada", 3);
+	Territory c = Territory(4, "Mexico", 3);
+	Territory c1 = Territory();
+	b.getTroopsSationed();//how many troops are stationed here
+	c1.setTroopsSationed(1);//set how many troops
+	b.addTroopsSationed(3);// add troops as reinforcemt or minus of troops as they die (troops = -1||-2)
+	cout<<b.toString();//return the state of the territory as a string
+	c1.setTerritoryId(6);
+	b.getTerritoryId();
+	c1.setTerritoryName("Quebec");
+	b.getTerritoryName();
+	//c1.setInContinent(3);
+	a.addAdjacent(&b);
+	a.addAdjacent(&c);
+	cout<<a.toString();
+	cout << c1.toString();
+	cout << "enter any char and then enter to exit";
+	char d1;
+	cin >> d1;
+	return 0;

```

Comment:
Same here, you just ignored everything I and Jeremy worked for from asg 1, it is already working as it is, there is no need to make a big changes. I am also relying on this on my part. Please work with what we have don't make big changes.

Reply:
@tramyardg  there is some of your code I do not understand 
In the map class 
1. you have a vector of pointers, where are the pointers deleted?
2. In the map destructor why is there the brackets by delete[] adjList, adjList is a pointer to a list of Territories we can just delete adjList.
3. The copy constructor and the = operator should be coping from m or t, why is it assuming m and t are the default values, what if we called functions to add things to it then called the constructor or equals operator?
4. Are we relying on default initialization for half of the constructors and what value and what happens if the constructor which does not have a destination (Territory(int src, string name, Continent c);) was used and then the command of addEdge from map was called on that edge (do we have any function to catch that and prevent it from crashing?
5.why is the getters (getSource() and getDestination()) declared as friend and not declared in the function (continent::getSource()...)?
6. if have constructors that do not take and initialize every value why do we not have setters so the values can be initialized later on?
7. And while truthfully we do not need to overload the equal operator for the continent and  Territory classes for the sake of the assignment we need them same for the copy constructors.

For the MapLoader 
1. why is the destructor empty, (where are you deleting the pointers in the vectors)?
2. why is the copy constructor and = operator making the file name "maps/canada.map" what if the file we are copying from has a diffrent file name?

Although this was not done by you I also have questions on this classes that maybe you can answer.
In the Player class.
1. why is the comment to do in lines 35-40 still there was have all of those in the class.
2. what is the purpose of this variable Player * thisPlayerRef = nullptr;
3. why is the playerID a const as if it is a const it must be initialized when it is declared?
4.  In the copy constructor when you have something like this (//Copy territories
	for (ptrTerritory = anotherPlayer.myTerritories->begin(); ptrTerritory != anotherPlayer.myTerritories->end(); ++ptrTerritory) {
		myTerritories->push_back(*ptrTerritory);
})
is this a deep or shallow copy?
5. why are there both issueOrder and addOrder when they seem to be doing the same thing?
6. why is the implementation of addMyTerritory and addTargetTerritory commented out?
7. why is there no removeMyTerritory and removeTargetTerritory functions?

I will admit that the cards class also has errors and I am in middle of fixing them, also my thoughts were anything that we anyway need to fix we should also simplify knowing from the FAQ that any non self defined functions do not need to be pointers (such as int, vector<>, and list<>.

Reply:
@tramyardg I will just assume you guys did correct and delete this comment as well as the earlier ones.